src/ts/components/subtitletogglebutton.ts

```diff
@@ -0,0 +1,63 @@
+import { ToggleButton, ToggleButtonConfig } from './togglebutton';
+import { i18n } from '../localization/i18n';
+import { PlayerAPI } from 'bitmovin-player';
+import { UIInstanceManager } from '../uimanager';
+import { StorageUtils } from '../storageutils';
+import { SubtitleSelectBox } from './subtitleselectbox';
+import { SettingsPanel } from './settingspanel';
+
+
+export class SubtitleToggleButton extends ToggleButton<ToggleButtonConfig> {
+    private settingsPanel: SettingsPanel;
+    private subtitleSelectBox: SubtitleSelectBox;
+    private player: PlayerAPI;
+
+    constructor(subtitleSettingsOpenButton: SettingsPanel, subtitleSelectBox: SubtitleSelectBox, config: ToggleButtonConfig = {}) {
+        super(config);
+
+        this.settingsPanel = subtitleSettingsOpenButton;
+        this.subtitleSelectBox = subtitleSelectBox;
+
+        const defaultConfig: ToggleButtonConfig = {
+            cssClass: 'ui-subtitletogglebutton',
+            onClass: 'subtitles-on',
+            offClass: 'subtitles-off',
+            text: i18n.getLocalizer('settings.subtitles'),
+            ariaLabel: i18n.getLocalizer('settings.subtitles'),
+        };
+
+        this.config = this.mergeConfig(config, defaultConfig as ToggleButtonConfig, this.config);
```

Comment:
default config should already be of type `ToggleButtonConfig`, since it's declared like this.
```suggestion
        this.config = this.mergeConfig(config, defaultConfig, this.config);
```
should do the trick.